### ISSUE TITLE
Bug 58167046: [React Native New Architecture]: Narrator remains silent on invoking the 'TouchableOpacity counter' button.

### ISSUE THREAD

**Issue by vineethkuttan** [microsoft/react-native-gallery#605]
**Build Details**:
OS Version: Dev (OS Build 27881.1000)
App: React Native Gallery Experimental
AT: Narrator

 

**Repro-Steps**:

1. Launch the React Native Gallery- Experimental app.
2. Navigate to the 'Expand menu' button and invoke it.
3. Navigate to 'All samples' button and invoke it.
4. Navigate to 'Touchable Opacity' button and invoke it.
5. Navigate to the 'TouchableOpcaity counter' button. Hit Enter to invoke it.
6. Observe the announcement and behavior.

**Actual Result**:
On invoking the 'TouchableOpacity counter' button, visually the value is getting changed but Narrator remains silent.

**Expected Result**:
On invoking the 'TouchableOpacity counter' button, Narrator should announce the visually appeared changes. It should not remains silent.

**User impact**:

Users with visual impairment who rely on screen reader will face difficulties if Narrator does not announce the change state information for buttons.

**MAS Reference**:  [MAS 4.2.12 – Change Notification](https://aka.ms/MAS4.2.12)

Trap ID: [1.4 - Uncomprehended Element](https://aka.ms/UncomprehendedElement)
 
“Have feedback on bugs logged, please tag bug as “A11yRCA” and add your feedback in the comment section.”
